### ISSUE TITLE
multiple monitors and panels

### ISSUE THREAD

**Issue by fossfreedom** [solus-project/budgie-desktop#436]
just trying using virtualbox and its multiple screen capability.

How do I get the top panel to display  across two screens?

![untitled](https://cloud.githubusercontent.com/assets/996240/15048683/7198a258-12e2-11e6-912e-70ef1e6bebfb.png)

as you can see settings - display shows the second screen and you can span applications across them.

The top panel though doesn't span across.

![untitled3](https://cloud.githubusercontent.com/assets/996240/15048792/e911c058-12e2-11e6-9d9c-76e516a6d49b.png)

**Comment by ikeydoherty** [comment]
Panel follows the primary monitor. What's the specific behaviour we're asking for here.. ? Note I don't have a multimonitor setup to test things, so I'll need help.

**Comment by fossfreedom** [comment]
The OP in question is querying whether you can specify one panel on one screen and a completely separate panel on a second screen. I guess something like if multiple display screens are detected then where you create a panel in raven you are given an option of which screen it appears on.

From my view point the work computer has one panel stretched across two monitors where the task list would eventually span across the second screen when you are working in lots of windows.

Yeah ... Nigh on impossible to implement on a single only monitor setup.  The screenshots show two virtual box displays. I've no idea how closely the multiple display capabilities of virtual box mimic real monitors.

**Comment by ikeydoherty** [comment]
Sorry to put a cloth in that exhaust pipe, but we're not going to support panning of displays. There are a billion and one things to consider, including, but not limited to, DPI differences, physical pixel spread, resolutions, as well as actual physical layout.

Individual panels on separate monitors, I'm happy to add that in future (We have the plumbing for it) - but no-go to panning

**Comment by mdsitton** [comment]
I don't know what the op wanted, but i would love to see the ability to put panels on secondary monitors(Spanning is almost always a horrible idea...)

**Comment by ikeydoherty** [comment]
Yeah I'm happy to implement per-monitor-panels, but just not spanning :)

**Comment by mdsitton** [comment]
Me and my 3 monitor setup thanks you :+1:

**Comment by fossfreedom** [comment]
Another variation on a theme here from a reddit user - same issue or a new one?

"The only options I see for moving the panel are "Top" and "Bottom". I run a 3 monitor setup and the middle monitor needs to be the "primary" display for fullscreen applications, but I want the left monitor to have the panel. How do I do this?"

**Comment by ikeydoherty** [comment]
That falls under the same category imo @fossfreedom

**Comment by Blquinn** [comment]
Did 'per-monitor-panels' ever get implemented? If so, how do I adjust that setting?

**Comment by ikeydoherty** [comment]
No, if they had we would've announced it.

**Comment by trallamochila** [comment]
I also hope to see panels on my both monitors. We will wait impatiently

**Comment by ikeydoherty** [comment]
You'll have to wait patiently, sorry :P It'll come for Budgie 11.

**Comment by ruttydm** [comment]
I also want a possibility for panels on a second monitor. There more people who switch to another desktop environment because of this.

**Comment by ikeydoherty** [comment]
Yes it's already planned @ruttydm - notice it's on the 11 milestone. Bumping or +1ing issues won't make them happen sooner :)

**Comment by astillich** [comment]
Given that 11 is probably not going to be released for some time, how much effort would it be to implement in current budgie? I might give it a try, if you could provide some rough instructions about what to do.

**Comment by ikeydoherty** [comment]
I don't want this in Budgie 10 tbqh about it. Budgie 11 for features, Budgie 10 is maintenance only/

**Comment by Subsparx** [comment]
+1 on needing this. I wanted to switch to Budgie from KDE but I don't think I can due to lack of this functionality. I want that bar on a monitor other than my primary, as I want it visible if I have say a video game open.

Additionally, if you have multiple monitors stacked vertically, it causes some super annoying bugs. I can't full screen an application on the top monitor without it leaving the bar for the panel worth of space up there, but if I full screen something on the bottom monitor, the top panel covers the title bar making it impossible to use the title bar for the application I just full screened. Completely unusable for a daily use system without the ability to move it.

We need something like the dock has. I can shift right click that and go to preferences and move it to any location on any monitor I want, which is what I would like, and I assume would be helpful for most others as well.

**Comment by dancespiele** [comment]
I have 3 monitors and it is really annoying to have to move to the main monitor if I want to run another application or change the desktop, etc... I would like to see the bottom main panel in my three monitors. Waiting for Budgie 11 to see it implemented :)

**Comment by RosieCode95** [comment]
+1 for this feature, its honestly the only reason i cant use the desktop ATM having 3+ screens and a large dev env just bloats the single menu-bar out.

if possible i would like to be able to have a menu bar on all screens and isolate the launch icons for just the screen that has the active window

**Comment by hsiW** [comment]
This is also something I'd really like, I've been stuck with other xfce and am wanting to make the switch but until i can have a panels on my other montiors, I won't be able to really use this as I rely on that feature a ton.

**Comment by Kattus** [comment]
Was really starting to like Budgie but not being able to have several panels is just unacceptable.
What I like to do is have two panels with the task list of each display separated, to make things a lot less messy.

**Comment by JoshStrobl** [comment]
This isn't something that is going to be addressed until our rewrite with Budgie 11. So no real need to necro this. We're aware that it is a much desired feature request.

**Comment by ajcollett** [comment]
So, when you have 3+ displays, you lose the option to mirror. Just want to check, should I open a new issue for that? Or is that covered by this?

In my setup, I have a dell XPS 13 that connects to 2 displays. That XPS then sits with it's lid closed (effectively only 2 displays in use) and then I currently use xrandr to clone the remaining displays. 

The flexibility to clone to some and extend to others might be too much, but maybe just being able to clone all?

**Comment by ruthan** [comment]
I want this too.. Make primary monitor - primary for all launched games, new icons.. and clone panel to all monitors, which all items shared.
  I was requested 4 years ago, without it, no great desktop experience possible..

**Comment by luuluuu** [comment]
+1
Such functionality is very necessary. I am trying to switch to budgie from Unity, but it is stopped by the inability to show the top bar on all monitors and the automatic detection of the active monitor. Also, of course, I would like to be able to semantic search based on zeitgeist, but this is another issue

**Comment by faedy2** [comment]
I just moved from GNOME to Budgie
Literally the one thing that I wish I could get is panels on multiple screens
Even if the panel would have to be duplicated, I'm fine with that

**Comment by rbreaves** [comment]
How did this one fall through the cracks? We apparently dissuaded anyone from working on this as well for the last 5 years because of Budgie 11? I may make this my next target to work on. I have no idea where Budgie 11 is at but I do want this feature.

**Comment by dlangerm** [comment]
Hopping on the train of wanting this feature. All I'd need/want is an option in the panel settings to show the panel on all monitors instead of just the primary. Just installed budgie yesterday and I really love it, and I'm glad this is a requested feature because like others have said it's basically my one big annoyance. Everything else is fantastic :smile:

**Comment by rbreaves** [comment]
An alternative right now is to install xfce4-panel and set it up to run on the 2nd display imo.

**Comment by joeldeteves** [comment]
The behaviour in Windows, and one I'd like to see in Budgie is to have multiple panels on multiple screens:
- Panel A -> Screen A
- Panel B -> Screen B

This is particularly useful in situations where you have the panel set to auto hide, which maximizes screen real estate. So rather than having to move over to the bottom edge of the left panel again, you can simply move your mouse down to the bottom edge of the screen you're currently on.

**Comment by bra1n** [comment]
It's 2025 now and it would still be nice to have one panel per screen as an option.
